andela/annie-hall-cfh
# #152798647  Add in-app gaming group chat

### What does this PR do?

It adds in-game chat feature to the application

### Description of Task to be completed?

* Design and implement in-app chat pane using scss
* Add chat pane controller that controls chat pane hide and show animations using jquery
* Make chat pane retractable
* Setup Firebase application console for storing chat messages
* Add function that use socket to recieve and emit messages to all other players in the game session
* Add notification chat icon container

### How should this be manually tested?

```
 1. From your terminal or console window, run brew install yarn
 2. Complete the basic setup of the project gotten from the README, on your local repository.
 3. Cd to the root of the project folder in your console.
 4. First run the command "yarn install" to install all required and updated node packages including
    bower components
 5. Log on to firebase and create a database. Make sure read and write permissions to the database 
     are enabled
 6. Create a .env file in root of project folder and add 
      MONGOHQ_URL=mongodb://anniehall:anniehall123@ds113826.mlab.com:13826/cfh_dev
      JWT_SECRET=anyRandomWordsOfYourChoose     
      RANDOM_CHARACTERS=anyRandomCharactersOfYourChoice
      API_KEY=yourFirebaseApiKey
      AUTH_DOMAIN=yourFirebaseAuthDomain
      DATABASE_URL=yourfirebaseDatabaseUrl
      PROJECT_ID=yourFirebaseProjectID
      STORAGE_BUCKET=yourFirebaseStorageBucket
      MESSAGING_SENDER_ID=yourFirebaseMessagingSenderID
    inside the file.
 7. Run the command `yarn global add gulp-cli`
 8. Run the command `yarn start` or `gulp build && gulp` to build and start the application.
 9. Navigate to your browser and enter localhost:3000
 10. Invite other players (atleast 3 or atmost 11) to your game session.
11. Click the chat icon container
12. Send message by clicking enter or the envelop icon after typing in a chat phrase or letter

```

Any background context you want to provide?

Nil

### What are the relevant pivotal tracker stories?

story type: feature
story Id: 152798647
story title: Users should be able to chat with the group while they are in a game

### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2017-12-14 at 3 39 44 pm" src="https://user-images.githubusercontent.com/22524619/34000178-d203ecf4-e0ec-11e7-878e-64d9ee2aed90.png">

<img width="1440" alt="screen shot 2017-12-14 at 3 40 37 pm" src="https://user-images.githubusercontent.com/22524619/34000200-e1339ce2-e0ec-11e7-832a-5f47bdc118ff.png">

<img width="1440" alt="screen shot 2017-12-14 at 3 41 02 pm" src="https://user-images.githubusercontent.com/22524619/34000218-ea7df43c-e0ec-11e7-8f21-ba1bc8a85f91.png">

<img width="1440" alt="screen shot 2017-12-14 at 3 43 11 pm" src="https://user-images.githubusercontent.com/22524619/34000241-f7ed3b1e-e0ec-11e7-880e-c51af47b6bd9.png">

<img width="1440" alt="screen shot 2017-12-14 at 3 46 10 pm" src="https://user-images.githubusercontent.com/22524619/34000280-1a6f2f3a-e0ed-11e7-88b6-7a8edf7a0f11.png">
